src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12588,51 +12580,49 @@ void func_8084FF7C(Player* this) {
     }
 }
 
-#ifdef NON_MATCHING
-// regalloc differences
 void func_8085002C(Player* this) {
-    s32 pad;
+    s16 v0 = D_80858AC8.unk_08;
+    s16 v1 = D_80858AC8.unk_06;
     s16 sp2A;
     s16 sp28;
     s16 sp26;
 
-    D_80858AC8.unk_06 -= D_80858AC8.unk_06 >> 3;
-    D_80858AC8.unk_08 -= D_80858AC8.unk_08 >> 3;
+    D_80858AC8.unk_06 -= v1 >> 3;
+    D_80858AC8.unk_08 -= v0 >> 3;
     D_80858AC8.unk_06 += -D_80858AC8.unk_00 >> 2;
     D_80858AC8.unk_08 += -D_80858AC8.unk_02 >> 2;
 
     sp26 = this->actor.posRot.rot.y - this->actor.shape.rot.y;
 
-    sp28 = this->actor.speedXZ * -200.0f * Math_Coss(sp26) * (Math_Rand_CenteredFloat(2.0f) + 10.0f);
-    sp2A = this->actor.speedXZ * 100.0f * Math_Sins(sp26) * (Math_Rand_CenteredFloat(2.0f) + 10.0f);
+    sp28 = (s32)(this->actor.speedXZ * -200.0f * Math_Coss(sp26) * (Math_Rand_CenteredFloat(2.0f) + 10.0f)) & 0XFFFF;
+    sp2A = (s32)(this->actor.speedXZ *  100.0f * Math_Sins(sp26) * (Math_Rand_CenteredFloat(2.0f) + 10.0f)) & 0XFFFF;
```

Comment:
I believe the changes in these 2 lines are the only thing necessary to match this one.
Also we use 0xFFFF, not 0XFFFF (not sure if the formatter would fix that on its own)